crates/daphne/src/roles/mod.rs

```diff
@@ -773,6 +773,60 @@ mod test {
 
     async_test_versions! { handle_agg_job_req_failure_hpke_decrypt_error }
 
+    #[tokio::test]
+    async fn handle_unknown_public_extensions() {
+        let version = DapVersion::Latest;
+        let t = Test::new(version);
+        let task_id = &t.time_interval_task_id;
+        let task_config = t.leader.unchecked_get_task_config(task_id).await;
+        let mut report = t.gen_test_report(task_id).await;
+        report.report_metadata.public_extensions = Some(vec![Extension::NotImplemented {
+            typ: 0x01,
+            payload: vec![0x01],
+        }]);
```

Comment:
Don't override, as this causes hpke decryption to fail. Here and below.

Reply:
I'm leaving the one below in place. The test below checks that a malformed report is appropriately handled. Generating the malformed report with the correct HPKE data would require a lot more code, and given that in this test the HPKE data is never checked this is much less brittle.